api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "goals", "verbose" })
+@EqualsAndHashCode
+public class KafkaClusterRebalanceSpec implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private List<String> goals;
```

Comment:
Presumably we can't just write arbitrary strings in here. How are they constrained? Are they chosen from a set of fixed string which come with CC? Are they pluggable? Even if we don't chose to constrain this using the CRD schema we should at least include in the `@Description` some of the supported goals, to make it a bit more obvious to the user what they need to supply here.

Reply:
You can find the list of goals here https://github.com/linkedin/cruise-control#goals. It's possible for a CC user writing its own goal library but it's something that we are not going to support right now. It should be something possible like we do for Kafka Connect: the user writes the goal, build a jar and should include it into a custom Docker image to use as image for the cruise control. Tbh I am fine on listing some of the goals in the description but I would not add any constraint on the CRD itself. If the user set not existing goals in the list, the operator will send the request to CC anyway but will get an error which will be reported in the status of the custom resource.